src/ring_theory/ideal/basic.lean

```diff
@@ -373,6 +375,67 @@ theorem is_prime.pow_mem_iff_mem {I : ideal α} (hI : I.is_prime)
   {r : α} (n : ℕ) (hn : 0 < n) : r ^ n ∈ I ↔ r ∈ I :=
 ⟨hI.mem_of_pow_mem n, (λ hr, I.pow_mem_of_mem hr n hn)⟩
 
+theorem pow_multiset_sum_mem_span_pow (s : multiset α) (n : ℕ) :
+  s.sum ^ (s.card * n + 1) ∈ span ((s.map (λ x, x ^ (n + 1))).to_finset : set α) :=
+begin
+  induction s using multiset.induction_on with a s hs,
+  { simp },
+  simp only [finset.coe_insert, multiset.map_cons, multiset.to_finset_cons, multiset.sum_cons,
+    multiset.card_cons, add_pow],
+  refine submodule.sum_mem _ _,
+  intros c hc,
+  rw mem_span_insert,
+  by_cases h : n+1 ≤ c,
+  { refine ⟨a ^ (c - (n + 1)) * s.sum ^ ((s.card + 1) * n + 1 - c) *
+      (((s.card + 1) * n + 1).choose c), 0, submodule.zero_mem _, _⟩,
+    rw mul_comm _ (a ^ (n + 1)),
+    simp_rw ← mul_assoc,
+    rw [← pow_add, add_zero, add_tsub_cancel_of_le h], },
+  { use 0,
+    simp_rw [zero_mul, zero_add],
+    refine ⟨_,_,rfl⟩,
+    replace h : c ≤ n := nat.lt_succ_iff.mp (not_le.mp h),
+    have : (s.card + 1) * n + 1 - c = s.card * n + 1 + (n - c),
+    { rw [add_mul, one_mul, add_assoc, add_comm n 1, ← add_assoc, add_tsub_assoc_of_le h] },
+    rw [this, pow_add],
+    simp_rw [mul_assoc, mul_comm (s.sum ^ (s.card * n + 1)), ← mul_assoc],
+    exact mul_mem_left _ _ hs }
+end
+
+theorem pow_sum_mem_span_pow {ι} (s : finset ι) (f : ι → α) (n : ℕ) :
```

Comment:
```suggestion
theorem sum_pow_mem_span_pow {ι} (s : finset ι) (f : ι → α) (n : ℕ) :
```
since the statement is `sum ^ ...` not `... ^ sum`

Reply:
I never remembered whether `pow_sum` meant the pow of sum or pow and then sum.

Reply:
For binary "operators" usually we have `a_foo_b` be a statement about `foo a b`